NuGetProtocolSamples/Program.cs

```diff
@@ -143,6 +153,60 @@ public static async Task SearchPackages()
             {
                 Console.WriteLine($"Found package {result.Identity.Id} {result.Identity.Version}");
             }
+#endregion
+        }
+
+        public static void CreatePackage()
+        {
+            // This code region is referenced by the NuGet docs. Please update the docs if you rename the region
+            // or move it to a different file.
+#region CreatePackage
```

Comment:
Creating a package that works well is full of gotchas. I'm not sure we should have a sample on how to use the NuGet SDK to create a package. I'd rather direct people to packing an SDK style project instead.  I do like the sample on reading the package though.

Reply:
I thought about that too but I thought erring on the side of transparency for a public API surface area is better. The API is public and committed to and therefore is usable today. Perhaps we can add a strongly worded warning saying "using the API to create packages is generally not recommend and should be using the tooling so that best practices are automatically applied"?

I don't feel super strongly about the inclusion of this sample but it does seem strange to have a committed public API with no mentions for, or against, it's usage. Including the sample with a warning is better than not talking about it and hoping people don't use it, right?

Reply:
If the consensus is to just not mention the API in SDK docs/samples, I'm okay with that too 😄. Just seems weird to me.

Reply:
I added a strongly worded message on the docs PR:
https://github.com/NuGet/docs.microsoft.com-nuget/pull/2126

> It is strongly recommended that NuGet packages are created using the official NuGet tooling and not using this low level API. There are a variety of characteristics important for a well-formed package and the latest version of tooling helps incorporate these best practices.
>
> For more information about creating NuGet packages, see the overview of the package creation workflow and the documentation for official pack tooling (e.g. using the dotnet CLI).

Reply:
Added the note to the code as well.